testing/python/approx.py

```diff
@@ -377,6 +377,22 @@ def test_bool(self):
 
         assert err.match(r"approx\(\) is not supported in a boolean context")
 
+    def test_list_with_str(self, assert_approx_raises_regex):
+        assert_approx_raises_regex(
+            [1.1, 2, "word"],
+            [1.0, 2, "different"],
+            [
+                "",
+                r"  comparison failed. Mismatched elements: 2 / 3:",
+                rf"  Max absolute difference: {SOME_FLOAT}",
+                rf"  Max relative difference: {SOME_FLOAT}",
+                r"  Index \| Obtained\s+\| Expected\s+",
+                r"\s*0\s*\|\s*1\.1\s*\|\s*1\.0\s*±\s*1\.0e\-06\s*",
+                r"\s*2\s*\|\s*word\s*\|\s*different\s*",
+            ],
+            verbosity_level=2,
+        )
```

Comment:
While at it, let's add an assert that works:

```suggestion
        )
        assert [1.1, 2, "word"] == pytest.approx([1.1, 2, "word"])
```